frontend/demo/src/actions/channel/index.ts

```diff
@@ -0,0 +1,78 @@
+import {createAction} from 'typesafe-actions';
+import _, {Dispatch} from 'redux';
+
+import {doFetchFromBackend} from '../../api/airyConfig';
+
+import {
+  Channel,
+  channelsMapper,
+  channelMapper,
+  ConnectChannelRequestPayload,
+  ExploreChannelRequestPayload,
+  DisconnectChannelRequestPayload,
+} from '../../model/Channel';
+
+const SET_CURRENT_CHANNELS = '@@channel/SET_CHANNELS';
+const ADD_CHANNELS = '@@channel/ADD_CHANNELS';
+
+export const setCurrentChannelsAction = createAction(SET_CURRENT_CHANNELS, resolve => (channels: Channel[]) =>
+  resolve(channels)
+);
+
+export const addChannelsAction = createAction(ADD_CHANNELS, resolve => (channels: Channel[]) => resolve(channels));
+
+export function getChannels() {
+  return async (dispatch: Dispatch<any>) => {
+    return doFetchFromBackend('channels.list')
+      .then(response => {
```

Comment:
We need to type the responses also. If we type them and the backend changes something it will automatically scream and won't lead to odd issues.